src/Nethermind/Nethermind.Synchronization/Blocks/BlockDownloader.cs

```diff
@@ -475,7 +485,13 @@ protected async Task RequestReceipts(PeerInfo peer, CancellationToken cancellati
                 for (int i = 0; i < result.Length; i++)
                 {
                     TxReceipt[] txReceipts = result[i];
-                    if (!context.TrySetReceipts(i + offset, txReceipts, out Block block))
+                    Block block = context.GetBlockByRequestIdx(i + offset);
+                    if (block.Header.HasBody && block.Transactions.Length == 0)
```

Comment:
what about block with only uncles?

Reply:
Removed. Moved to another PR. That was probably the cause of the issue on the goerli sync?

Reply:
Removed. Moved to another PR. That was probably the cause of the issue on the goerli sync?